src/internal/m365/collection/drive/delta_tree.go

```diff
@@ -289,19 +292,11 @@ func (face *folderyMcFolderFace) addFile(
 	face.fileIDToParentID[id] = parentID
 	parent.files[id] = lastModifed
 
+	delete(face.deletedFileIDs, id)
```

Comment:
sanity check: if we're removing from the deleted files map here, is generating the prefix matcher output at the end going to walk both the set of live items and the set of explicitly deleted items?